components/core/Application.js

```diff
@@ -424,6 +471,99 @@ export default class ApplicationPage extends React.Component {
     this._handleRegisterLoadingFinished({ keys });
   };
 
+  _handleOptimisticUpload = async ({ files, slate }) => {
+    let optimisticFiles = [];
+
+    for (let i = 0; i < files.length; i++) {
+      let id = uuid();
+      let dataURL = await this._handleLoadDataURL(files[i]);
+      let data = {
+        id,
+        filename: files[i].name,
+        data: {
+          name: files[i].name,
+          type: files[i].type,
+          size: files[i].size,
+        },
+        decorator: "OPTIMISTIC-FILE",
```

Comment:
Instaed of using decorator = "OPTIMISTIC-FILE" maybe we can do something like file.optimistic = true 

Decorator might be used for other things so we'd like to reserve it for that ideally. Plus it's easier to check for (`if (file.optimistic)`) and b/c with `decorator.startsWith("OPTIMISTIC")` you have to worry about keeping capitalization consistent